GitUI/Avatars/GravatarProvider.cs

```diff
@@ -0,0 +1,88 @@
+﻿using System;
+using System.Drawing;
+using System.Security.Cryptography;
+using System.Text;
+using System.Threading.Tasks;
+using GitCommands;
+using JetBrains.Annotations;
+
+namespace GitUI.Avatars
+{
+    public sealed class GravatarProvider : IAvatarProvider
+    {
+        // For details about the rating see https://en.gravatar.com/site/implement/images#rating
+        // "g": suitable for display on all websites with any audience type.
+        private const string _rating = "g";
+
+        private readonly IAvatarDownloader _downloader;
+        private readonly AvatarFallbackType? _fallback;
+        private readonly bool _forceFallback;
+
+        public GravatarProvider(
+            [NotNull] IAvatarDownloader downloader,
+            AvatarFallbackType? fallback,
+            bool forceFallback = false)
+        {
+            _downloader = downloader ?? throw new ArgumentNullException(nameof(downloader));
+            _fallback = fallback;
+            _forceFallback = forceFallback;
+        }
+
+        public static bool IsFallbackSupportedByGravatar(AvatarFallbackType fallback)
+        {
+            return SerializeFallbackType(fallback) != null;
+        }
+
+        /// <inheritdoc/>
+        public Task<Image> GetAvatarAsync([NotNull] string email, string name, int imageSize)
```

Comment:
Why not async?

Reply:
It is async, but only passed through the task from this call: `return _downloader.DownloadImageAsync(avatarUri);` making this async and awaiting the call would just cause overhead, because the compiler would build an async-await state machine for a method where it is not needed.